test/org/zaproxy/zap/extension/pscanrules/XFrameOptionsScannerUnitTest.java

```diff
@@ -0,0 +1,392 @@
+        rule.setLevel(AlertThreshold.HIGH);
+
+        HttpMessage msg = new HttpMessage();
+        msg.setRequestHeader("GET http://www.example.com/test/ HTTP/1.1");
+        
+        msg.setResponseBody("<html></html>");
+        msg.setResponseHeader(
+                "HTTP/1.1 401 OK\r\n" +
+                "Server: Apache-Coyote/1.1\r\n" +
+                "Content-Type: text/html;charset=ISO-8859-1\r\n" +
+                "Content-Length: " + msg.getResponseBody().length() + "\r\n");
+        String response = msg.getResponseHeader().toString() + msg.getResponseBody().toString();
+        Source source = new Source(response);
+        rule.scanHttpResponseReceive(msg, -1, source);
+
+        assertThat(alertsRaised.size(), equalTo(0));
+    }
+
+    @Test
+    public void noXframeOptionsPlainTextLow () throws HttpMalformedHeaderException {
+        
+        rule.setLevel(AlertThreshold.LOW);
+
+        HttpMessage msg = new HttpMessage();
+        msg.setRequestHeader("GET http://www.example.com/test/ HTTP/1.1");
+        
+        msg.setResponseBody("<html></html>");
+        msg.setResponseHeader(
+                "HTTP/1.1 401 OK\r\n" +
+                "Server: Apache-Coyote/1.1\r\n" +
+                "Content-Type: text/plain;charset=ISO-8859-1\r\n" +
+                "Content-Length: " + msg.getResponseBody().length() + "\r\n");
+        String response = msg.getResponseHeader().toString() + msg.getResponseBody().toString();
+        Source source = new Source(response);
+        rule.scanHttpResponseReceive(msg, -1, source);
+
+        assertThat(alertsRaised.size(), equalTo(1));
+        assertThat(alertsRaised.get(0).getParam(), equalTo(HttpHeader.X_FRAME_OPTION));
+        assertThat(alertsRaised.get(0).getEvidence(), equalTo(""));
+    }
+
+    @Test
+    public void noXframeOptionsPlainTextMed () throws HttpMalformedHeaderException {
+        
+        rule.setLevel(AlertThreshold.MEDIUM);
+
+        HttpMessage msg = new HttpMessage();
+        msg.setRequestHeader("GET http://www.example.com/test/ HTTP/1.1");
+        
+        msg.setResponseBody("<html></html>");
+        msg.setResponseHeader(
+                "HTTP/1.1 401 OK\r\n" +
+                "Server: Apache-Coyote/1.1\r\n" +
+                "Content-Type: text/plain;charset=ISO-8859-1\r\n" +
+                "Content-Length: " + msg.getResponseBody().length() + "\r\n");
+        String response = msg.getResponseHeader().toString() + msg.getResponseBody().toString();
+        Source source = new Source(response);
+        rule.scanHttpResponseReceive(msg, -1, source);
+
+        assertThat(alertsRaised.size(), equalTo(0));
+    }
+
+    @Test
+    public void noXframeOptionsPlainTextHigh () throws HttpMalformedHeaderException {
+        
+        rule.setLevel(AlertThreshold.HIGH);
+
+        HttpMessage msg = new HttpMessage();
+        msg.setRequestHeader("GET http://www.example.com/test/ HTTP/1.1");
+        
+        msg.setResponseBody("<html></html>");
```

Comment:
Function name and content-type suggest text...while I suppose this isn't wrong (the text could be html tags), it's kind of misleading.